File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]):
        i, j = 0, len(mat) - 1
        res = 0
        while i < len(mat):
            res += mat[i][i]
            if i != j:
                res += mat[i][j]
            i += 1
            j -= 1
        return res

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.go
func diagonalSum(mat [][]int) int {
    i, j := 0, len(mat) - 1
    res := 0
    for i < len(mat){
        res += mat[i][i]
        if i != j{
            res += mat[i][j]
        }
        i += 1
        j -= 1
    }
    return res
}